src/typechecker.erl

```diff
@@ -1041,6 +1045,9 @@ expect_list_type(Ty, _, _) ->
 rewrite_list_to_nonempty_list({type, Ann, T, [ElemTy]})
   when T == list orelse T == nonempty_list ->
     {type, Ann, nonempty_list, [ElemTy]};
+rewrite_list_to_nonempty_list({type, Ann, T, []})
+  when T == string orelse T == nonempty_string ->
```

Comment:
This was a bit more tricky. The denormalized `string()` came from https://github.com/josefs/Gradualizer/blob/b09458159e72ef16598f4ff8499ef64a2a816061/src/typechecker.erl#L4142 since when typechecking `{match, _, _, _}` `add_type_pat` is called consecutively - https://github.com/josefs/Gradualizer/blob/b09458159e72ef16598f4ff8499ef64a2a816061/src/typechecker.erl#L4218-L4221 - and output of the first call is used as input to the second one. If `Pat2` happened to be a string, then we would've gotten a denormalized `string()` type back. Fixed in the last commit.